src/qibolab/transpilers/general_connectivity.py

```diff
@@ -0,0 +1,352 @@
+import random
+from copy import deepcopy
+
+import matplotlib.pyplot as plt
+import networkx as nx
+import numpy as np
+import sympy
+from qibo import gates
+from qibo.models import Circuit
+
+
+class Transpiler:
+    def __init__(self, connectivity="21_qubits", init_method="greedy", init_samples=100):
+        self.connectivity = self.set_connectivity(connectivity)
+        self.init_method = init_method
+        self.init_samples = init_samples
+
+    def custom_qubit_mapping(self, map):
+        """Define initial qubit mapping using a dictionary of the type {q0: 1, q1: 2, q2: 0}"""
+        self.init_method = "custom"
+        self.mapping = map
+
+    def set_connectivity(self, connectivity):
```

Comment:
Usually it is cleaner to use [Python's setters](https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters) in the place of `set_*`. I know we have them in a few places around qibolab but we should make an effort to clean this up and try to avoid it in new code. For example here something like
```suggestion
    @property
    def connectivity(self):
        return self._connectivity
        
    @connectivity.setter
    def connectivity(self, connectivity):
        ...
```
could work.